cin_validator/rules/cin2022_23/rule_4012Q.py

```diff
@@ -0,0 +1,57 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import rule_definition, CINTable, RuleContext
+from cin_validator.rule_engine import IssueLocator
+from cin_validator.test_engine import run_rule
+
+CINplanDates = CINTable.CINplanDates
+CINPlanStartDate = CINplanDates.CINPlanStartDate
+
+# define characteristics of rule
+@rule_definition(
+    code="4012Q",
+    module=CINTable.CINplanDates,
+    message="CIN Plan shown as starting and ending on the same day - please check",
+    affected_fields=[CINPlanStartDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    df = data_container[CINplanDates]
+    """
+    Within a <CINPlanDates> group, <CINPlanStartDate> (N00689) should not be the same as the <CINPlanEndDate> (N00690)
+    """
+    
+    df = df[df['CINPlanStartDate'] == df['CINPlanEndDate']]
+
+    failing_indices = df.index
+
+    rule_context.push_issue(
+        table=CINplanDates, field=CINPlanStartDate, row=failing_indices
+    )
```

Comment:
Ideally, when multiple columns are compared, we want to be able to push multiple columns in a way that shows the relationship between them. 
I'll put this on hold while we finalize plans on how to do that.